internal/tfdiags/consolidate_warnings.go

```diff
@@ -56,21 +61,30 @@ func (diags Diagnostics) Consolidate(threshold int, level Severity) Diagnostics
 
 		desc := diag.Description()
 		summary := desc.Summary
-		if g, ok := diagnosticGroups[summary]; ok {
+		var consolidationKey string
+		// If the diagnostic has a keyable extra info, use it as the consolidation key,
+		// otherwise use the summary.
```

Comment:
As @apparentlymart [suggested](https://github.com/opentofu/opentofu/issues/2751#issuecomment-2851394128), we should use both, the key and the summary to generate the `consolidationKey`.